pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -490,70 +490,72 @@ func (r *ReconcileHyperConverged) ensureCDI(instance *hcov1alpha1.HyperConverged
 	}
 	objectreferencesv1.SetObjectReference(&instance.Status.RelatedObjects, *objectRef)
 
-	// Handle CDI resource conditions
-	if found.Status.Conditions == nil {
-		logger.Info("CDI's resource is not reporting Conditions on it's Status")
-		conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-			Type:    conditionsv1.ConditionAvailable,
-			Status:  corev1.ConditionFalse,
-			Reason:  "CDIConditions",
-			Message: "CDI resource has no conditions",
-		})
-		conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-			Type:    conditionsv1.ConditionProgressing,
-			Status:  corev1.ConditionTrue,
-			Reason:  "CDIConditions",
-			Message: "CDI resource has no conditions",
-		})
-		conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-			Type:    conditionsv1.ConditionUpgradeable,
-			Status:  corev1.ConditionFalse,
-			Reason:  "CDIConditions",
-			Message: "CDI resource has no conditions",
-		})
-	} else {
-		for _, condition := range found.Status.Conditions {
-			// convert the CDI condition type to one we understand
-			switch conditionsv1.ConditionType(condition.Type) {
-			case conditionsv1.ConditionAvailable:
-				if condition.Status == corev1.ConditionFalse {
-					logger.Info("CDI is not 'Available'")
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionAvailable,
-						Status:  corev1.ConditionFalse,
-						Reason:  "CDINotAvailable",
-						Message: fmt.Sprintf("CDI is not available: %v", string(condition.Message)),
-					})
-				}
-			case conditionsv1.ConditionProgressing:
-				if condition.Status == corev1.ConditionTrue {
-					logger.Info("CDI is 'Progressing'")
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionProgressing,
-						Status:  corev1.ConditionTrue,
-						Reason:  "CDIProgressing",
-						Message: fmt.Sprintf("CDI is progressing: %v", string(condition.Message)),
-					})
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionUpgradeable,
-						Status:  corev1.ConditionFalse,
-						Reason:  "CDIProgressing",
-						Message: fmt.Sprintf("CDI is progressing: %v", string(condition.Message)),
-					})
-				}
-			case conditionsv1.ConditionDegraded:
-				if condition.Status == corev1.ConditionTrue {
-					logger.Info("CDI is 'Degraded'")
-					conditionsv1.SetStatusCondition(&r.conditions, conditionsv1.Condition{
-						Type:    conditionsv1.ConditionDegraded,
-						Status:  corev1.ConditionTrue,
-						Reason:  "CDIDegraded",
-						Message: fmt.Sprintf("CDI is degraded: %v", string(condition.Message)),
-					})
-				}
-			}
-		}
-	}
+	// TODO: Handle CDI resource conditions after https://github.com/kubevirt/containerized-data-importer/pull/910 merges
+
+	// // Handle CDI resource conditions
+	// if found.Status.Conditions == nil {
```

Comment:
@djzager my env is conveniently not getting past network, but I've seen this issue downstream stream.  What I think is happening is CDI's conditions are nil, so they are always getting set to !Available, Progressing, and Degraded.  Therefore, the HCO never gets to 1/1 Ready.

Reply:
weird. let me bring up an environment real quick (famous last words).